lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -273,11 +252,13 @@ class MailboxDashBoardController extends ReloadableController {
 
   void setSelectedMailbox(PresentationMailbox? newPresentationMailbox) {
     selectedMailbox.value = newPresentationMailbox;
+    searchCtrl.updateFilterEmail(mailbox: selectedMailbox.value);
```

Comment:
`SearchController` should observe that value from `dashboard` to follow single truth